server/testdb/fs/fsdb.go

```diff
@@ -0,0 +1,250 @@
+	if err != nil {
+		err = fmt.Errorf("could not get machineID: %w", err)
+		return
+	}
+	id = id[:10] // limit lenght to avoid issues with GA
+
+	if err != nil {
+		err = fmt.Errorf("could not get machineID: %w", err)
+		return
+	}
+	err = td.config().write("serverID", id)
+	if err != nil {
+		err = fmt.Errorf("could not save serverID into DB: %w", err)
+		return
+	}
+
+	return
+}
+
+type fileDB[K comparable, V any] struct {
+	path string
+}
+
+func (fdb fileDB[K, V]) read() (map[K]V, error) {
+	b, err := os.ReadFile(fdb.path)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			// allow working with not existing dbs
+			return nil, nil
+		}
+
+		return nil, err
+	}
+
+	var contents map[K]V
+	err = json.Unmarshal(b, &contents)
+	if err != nil {
+		return nil, err
+	}
+
+	return contents, nil
+}
+
+func (fdb fileDB[K, V]) get(key K) (v V, err error) {
+	rows, err := fdb.read()
+	if err != nil {
+		err = fmt.Errorf("cannot read db %s: %w", fdb.path, err)
+		return
+	}
+
+	row, exists := rows[key]
+	if !exists {
+		err = testdb.ErrNotFound
+		return
+	}
+
+	return row, nil
+}
+
+func (fdb fileDB[K, V]) write(key K, val V) error {
+	db, err := fdb.read()
+	if err != nil {
+		return err
+	}
+	if db == nil {
+		db = make(map[K]V)
+	}
+
+	db[key] = val
+
+	encoded, err := json.Marshal(db)
+	if err != nil {
+		return err
+	}
+
+	return os.WriteFile(fdb.path, encoded, 0644)
+}
+
+type file struct {
+	path string
+	info os.FileInfo
+}
+
+func (f file) write(in yaml.File) error {
+	b, err := in.Encode()
+	if err != nil {
+		return fmt.Errorf("cannot encode input for file %s: %w", f.path, err)
+	}
+
+	os.WriteFile(f.path, b, 0644)
+	if err != nil {
+		return fmt.Errorf("cannot write file %s: %w", f.path, err)
+	}
+
+	return nil
+}
+
+func (f file) readYaml() (yaml.File, error) {
+	b, err := os.ReadFile(f.path)
+	if err != nil {
+		return yaml.File{}, fmt.Errorf("cannot read file %s: %w", f.path, err)
+	}
+
+	yf, err := yaml.Decode(b)
+	if err != nil {
+		return yaml.File{}, fmt.Errorf("cannot decode file %s: %w", f.path, err)
+	}
+
+	return yf, nil
+}
+
+func (f file) readEnv() (map[string]string, error) {
+	b, err := os.ReadFile(f.path)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read file %s: %w", f.path, err)
+	}
+	return godotenv.Unmarshal(string(b))
+
+}
+
+func (f file) isYaml() bool {
+	return slices.Contains([]string{".yaml", ".yml"}, filepath.Ext(f.path))
+}
+
+func (f file) isEnv() bool {
+	return slices.Contains([]string{".env"}, filepath.Ext(f.path))
+}
+
+func (td *fsDB) getFiles() ([]file, error) {
+	return td.readDir(td.root)
+}
+
+func (td *fsDB) readDir(path string) ([]file, error) {
+	var files []file
+	err := filepath.Walk(path,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			files = append(files, file{path, info})
+			return nil
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	return files, nil
+}
+
+func (td *fsDB) Drop() error {
+	panic("not implemented")
```

Comment:
I think it's worth implementing this because it's very simple:

```suggestion
	return os.RemoveAll(td.root)
```